dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ParseConfigUtils.java

```diff
@@ -40,13 +40,20 @@
 
   public static final Map<String, Object> flatItems = new HashMap<>();
 
+  public static String CURRENT_VERSION_INFO = "default";
+
   private UpdateHandler updateHandler;
 
   public ParseConfigUtils(UpdateHandler updateHandler) {
     this.updateHandler = updateHandler;
   }
 
   public void refreshConfigItems(Map<String, Map<String, Object>> remoteItems) {
+
+    CURRENT_VERSION_INFO = remoteItems.getOrDefault("revision",new HashMap<>()).getOrDefault("version","default").toString();
+    //make sure the CURRENT_VERSION_INFO != ""
+    CURRENT_VERSION_INFO = CURRENT_VERSION_INFO.equals("") ? "default" : CURRENT_VERSION_INFO;
+    remoteItems.remove("revision");//the key revision is not the config setting
```

Comment:
How about users have a key named revision?

Reply:
it doesn't matter,as you know, if the user have a key named revision, it is shown as bellow
{
    "springmvc@springmvctest": {
        "revision": "gg"
    },
    "springmvc@springmvctest#0.0.3": {
        "revision": "gggg"
    },
    "revision": {
        "version": "default44-47"
    }
}